src/views/templates/list/hooks/useVirtualMachineTemplatesFilters.ts

```diff
@@ -24,39 +24,71 @@ const useTemplateProviders = (): TemplateProviderProps[] => {
   ];
 };
 
-// return VM template provider name or 'Other' if the provider name not included in the array of providers available for filtering
-const getTemplateProviderNameWithOther = (template: V1Template, templateProviders): string => {
-  const providerName = getTemplateProviderName(template);
+const useOsNames = () => {
+  const { t } = useKubevirtTranslation();
 
-  return !templateProviders?.find((s: TemplateProviderProps) => s.id === providerName) ||
-    providerName === 'Other'
-    ? 'Other'
-    : providerName;
+  return [
+    {
+      id: OS_NAME_TYPES.rhel,
+      title: OS_NAME_LABELS.rhel,
+    },
+    {
+      id: OS_NAME_TYPES.fedora,
+      title: OS_NAME_LABELS.fedora,
+    },
+    {
+      id: OS_NAME_TYPES.centos,
+      title: OS_NAME_LABELS.centos,
+    },
+    {
+      id: OS_NAME_TYPES.windows,
+      title: OS_NAME_LABELS.windows,
+    },
+    {
+      id: OS_NAME_TYPES.other,
+      title: t('Other'),
+    },
+  ];
 };
 
+// return the name (of VM template provider/OS) or 'Other' if the name not included in the array of available items for filtering
+const getItemNameWithOther = (itemName: string, items: ItemsToFilterProps[]): string =>
+  !items?.find((s: ItemsToFilterProps) => s.id === itemName) || itemName === 'Other'
+    ? 'Other'
+    : itemName;
+
 const includeFilter = (
   compareData: FilterValue,
-  providers: TemplateProviderProps[],
-  template: V1Template,
+  items: ItemsToFilterProps[],
+  itemName: string,
 ): boolean => {
-  const compareString = getTemplateProviderNameWithOther(template, providers);
+  const compareString = getItemNameWithOther(itemName, items);
 
   return compareData.selected?.length === 0 || compareData.selected?.includes(compareString);
 };
 
 const useVirtualMachineTemplatesFilters = (): RowFilter[] => {
   const { t } = useKubevirtTranslation();
   const providers = useTemplateProviders();
+  const osNames = useOsNames();
```

Comment:
this actually return a constant ( + the others translated string)
do you actually need a use method here ?
```suggestion
const OS_NAMES = [
    {
      id: OS_NAME_TYPES.rhel,
      title: OS_NAME_LABELS.rhel,
    },
    {
      id: OS_NAME_TYPES.fedora,
      title: OS_NAME_LABELS.fedora,
    },
    {
      id: OS_NAME_TYPES.centos,
      title: OS_NAME_LABELS.centos,
    },
    {
      id: OS_NAME_TYPES.windows,
      title: OS_NAME_LABELS.windows,
    },
    {
      id: OS_NAME_TYPES.other,
      title: t('Other'),
    },
  ];
  
  const osNames = [
      ...OS_NAMES,
      {
          id: OS_NAME_TYPES.other,
          title: t('Other'),
      },
];
```